api/v1/helpers/nouns/subjects.js

```diff
@@ -115,7 +115,7 @@ function traverseHierarchy(res) {
     }
 
     // filtered array is attached to the children key
-    res.children = filteredChildrenArr;
+    res.children = filteredChildrenArr.length ? filteredChildrenArr : undefined;
```

Comment:
is it better to make this undefined or an empty zero-length array?

Reply:
When the hierarchy comes from the db, subjects with no children have an undefined "children" attribute, instead of an empty array. This was changed for consistency with that.